### PR TITLE
Implementation of vast parser

### PR DESCRIPTION
**Pull request by gunhansancar**

### CODE REVIEW

File: MoyaNetworkDataSource.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
+//
+//  MoyaNetworkDataSource.swift
+//  SuperAwesome
+//
+//  Created by Gunhan Sancar on 06/07/2020.
+//
+
+import Alamofire
+
+class MoyaNetworkDataSource {
```
Comment (devgabrielcoman) [inline MoyaNetworkDataSource.swift:10]: Is this Moya or AF?

File: Pod/Plugin/Core/Classes/Components/HtmlFormatter.swift [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        
+        if let clickUrl = ad.creative.click_url {
+            return "<a href='\(clickUrl)' target='_blank'>\(img)</a>_MOAT_"
+        }
+        
+        return "\(img)_MOAT_"
+    }
+    
+    func formatRichMediaIntoHtml(_ ad: Ad) -> String {
+        let url = "\(ad.creative.details.url)?placement=\(1)&line_item=\(ad.line_item_id)&creative=\(ad.creative.id)&rnd=\(numberGenerator.nextIntForCache())"
```
Comment (devgabrielcoman) [inline Pod/Plugin/Core/Classes/Components/HtmlFormatter.swift:35]: placement=1 .... is that okay?

File: Pod/Plugin/Core/Classes/Model/VastModels.swift [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+//
+//  Created by Gunhan Sancar on 06/07/2020.
+//
+
+class VastAd {
+    var url: String?
+    var redirect: String?
+    var type: VastType = .Invalid
+    var events: [VastEvent] = []
+    var medias: [VastMedia] = []
```
Comment (devgabrielcoman) [inline Pod/Plugin/Core/Classes/Model/VastModels.swift:13]: media is already a plural, btw :P

File: Pod/Plugin/Core/Classes/Repositories/AdRepository.swift [hunk 4]
```diff
@@ -6,21 +6,32 @@
 //
 
 protocol AdRepositoryType {
-    func getAd(placementId: Int, request: AdRequest, completion: @escaping Completion<Ad>)
+    func getAd(placementId: Int, request: AdRequest, completion: @escaping OnResultListener<Ad>)
```
Comment (devgabrielcoman) [inline Pod/Plugin/Core/Classes/Repositories/AdRepository.swift:9]: Are you married to `OnResultListener` as a name? For whatever reason `OnResult` seems like a simpler / more concise name to me.

File: Pod/Plugin/Core/Classes/Repositories/AdRepository.swift [hunk 5]
```diff
@@ -6,21 +6,32 @@
...
     
-    func getAd(placementId: Int, request: AdRequest, completion: @escaping Completion<Ad>) {
+    func getAd(placementId: Int, request: AdRequest, completion: @escaping OnResultListener<Ad>) {
         dataSource.getAd(placementId: placementId,
-                         query: adQueryMaker.makeAdQuery(request),
-                         completion: completion)
+                         query: adQueryMaker.makeAdQuery(request)) { result in
+                            if case .success(let ad) = result {
+                                self.adProcessor.process(ad) { response in
+
```
Comment (devgabrielcoman) [inline Pod/Plugin/Core/Classes/Repositories/AdRepository.swift:30]: What happens here?

Comment (gunhansancar) [reply Pod/Plugin/Core/Classes/Repositories/AdRepository.swift:30]: That part will be implemented with the other UI parts as this PR only(mostly) about VAST for now.

**Pull request merged**